src/test/Union.fbs

```diff
@@ -0,0 +1,32 @@
+table Gemstone {
+  shine:double;
+}
+
+table Arms {
+  count:uint8;
+}
+
+table Skull {
+  name:string;
+}
+
+union ShieldDecorator { Gemstone, Arms, Skull }
+
+table Sword {
+  name:string;
+  damage:short;
+}
+
+table Shield {
+  protection:float;
+  primary_decorator: ShieldDecorator;
```

Comment:
nit: for consistency (I think no space is common style for flatbuffers? it's harder to read IMO but 🤷)
```suggestion
  primary_decorator:ShieldDecorator;
```